File: ContainerWithMostWater/two_pointers.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        most = (r - l) * min([height[l], height[r]])

        while l < r:
            curr = (r - l) * min([height[l], height[r]]) 
            most = max(most, curr)

            if height[l] > height[r]:
                r -= 1
            else:
                l += 1
        
        return most
        

File: ContainerWithMostWater/two_pointers.java
public class Solution {
    public int maxArea(int[] height) {
        var l = 0;
        var r = height.length - 1;
        var most = (r - l) * Math.min(height[l], height[r]);

        while (l < r) {
            var curr = (r - l) * Math.min(height[l], height[r]);
            most = Math.max(most, curr);
            if (height[l] > height[r]) {
                r--;
            } else {
                l++;
            }
        }

        return most;
    }
}
